desktop-widgets/filterwidget2.cpp

```diff
@@ -52,21 +55,16 @@ FilterWidget2::FilterWidget2(QWidget* parent) :
 	connect(ui.minVisibility, &StarWidget::valueChanged,
 		this, &FilterWidget2::updateFilter);
 
-	// We need these insane casts because Qt decided to function-overload some of their signals(!).
-	// QDoubleSpinBox::valueChanged() sends double and QString using the same signal name.
-	// QComboBox::currentIndexChanged() sends int and QString using the same signal name.
-	// Qt 5.7 provides a "convenient" helper that hides this, but only if compiling in C++14
-	// or higher. One can then write: "QOverload<double>(&QDoubleSpinBox::valueChanged)", etc.
-	connect(ui.maxAirTemp, static_cast<void (QDoubleSpinBox::*)(double)>(&QDoubleSpinBox::valueChanged),
+	connect(ui.maxAirTemp, QOverload<double>::of(&QDoubleSpinBox::valueChanged),
 		this, &FilterWidget2::updateFilter);
 
-	connect(ui.minAirTemp, static_cast<void (QDoubleSpinBox::*)(double)>(&QDoubleSpinBox::valueChanged),
+	connect(ui.minAirTemp, QOverload<double>::of(&QDoubleSpinBox::valueChanged),
```

Comment:
These are a distinct improvement, though. The `static_cast<>` trick was crazy (and probably undefined behavior).

Reply:
Completely agree